src/colmap/estimators/alignment.cc

```diff
@@ -408,9 +408,10 @@ bool AlignReconstructionsViaPoints(const Reconstruction& src_reconstruction,
   return report.success;
 }
 
-bool MergeReconstructions(const double max_reproj_error,
-                          const Reconstruction& src_reconstruction,
-                          Reconstruction* tgt_reconstruction) {
+bool MergeReconstructions(
+    const double max_reproj_error,
+    const Reconstruction& src_reconstruction,
+    const std::shared_ptr<Reconstruction>& tgt_reconstruction) {
```

Comment:
I suggest to make this a ref instead of a shared_ptr. The current shared_ptr interface enforces a particular memory management model on the user side. For example, a reconstruction in a unique_ptr would be difficult to pass here without copying the data.

Reply:
Good point. Fixed. For consistency, I have also updated `IncrementalTriangulator::reconstruction_` to be a reference.